src/translate.py

```diff
@@ -4856,13 +4885,17 @@ def make_arg(offset: int, type: Type) -> PassedInArg:
     for slot in abi.arg_slots:
         stack_info.add_known_param(slot.offset, slot.name, slot.type)
```

Comment:
I think there's some interaction in `add_known_param` that was broken here: the diff for MM with context is very large and seems to use the original `Actor *` type for the first argument instead of the derived struct? 

[First 10k lines of diff](https://gist.github.com/zbanks/a86da755f7e7f076a9b6584f7b10da90), `ArrowFire_Init` is a good example.

The `add_known_param` function is a little weird, and has a way to override `slot.type` for the arg. (I guess the `type` of the `PassedInArg` wasn't used before?)

Reply:
If you need some context files, [here are the ones I use for OOT & MM](https://gist.github.com/zbanks/537c38bee61f8fec1c67b01782cb4738).

Reply:
Good catch, serves me right for not running mm tests. I made it return the type to use for the parameter. (I wonder if that machinery could change to use the newly introduced arg temps, instead of having special handling in if_statements.py.) New diff, for both mm and oot this time: https://gist.github.com/simonlindholm/95b45d1096a964a67bb89088d4e0581b Looks like a slightly improvement, but completely by accident: I don't know what actually triggers the changes.